core/src/main/scala/akka/kafka/ProducerMessage.scala

```diff
@@ -35,8 +35,10 @@ object ProducerMessage {
    * `offset` of the produced message.
    */
   final case class Result[K, V, PassThrough](
-    offset: Long,
-    message: Message[K, V, PassThrough]
-  )
+      metadata: RecordMetadata,
+      message: Message[K, V, PassThrough]
+  ) {
+    def offset = metadata.offset()
```

Comment:
Ok :) In my opinion it's better to trust Kafka API as we did with exposing `ConsumerRecord`.